src/core/qgsbrowsermodel.cpp

```diff
@@ -183,12 +183,14 @@ QMap<QString, QgsDirectoryItem *> QgsBrowserModel::driveItems() const
   return mDriveItems;
 }
 
+
 void QgsBrowserModel::initialize()
 {
   if ( ! mInitialized )
   {
     connect( QgsProject::instance(), &QgsProject::readProject, this, &QgsBrowserModel::updateProjectHome );
     connect( QgsProject::instance(), &QgsProject::projectSaved, this, &QgsBrowserModel::updateProjectHome );
+    connect( QgsProject::instance(), &QgsProject::cleared, this, &QgsBrowserModel::updateProjectHome );
```

Comment:
I still think that this should all be done via the `QgsProject::homePathChanged`. I.e. the signal should be emitted from the `QgsProject::clear()` method (or `mHomePath.clear()` replaced with `setHomePath( QString() )` ).